File: DepartmentManagementModels/Department.cs
using System;
using System.Collections.Generic;

namespace DepartmentsAndEmployeesModels
{
    public class Department
    {
        public long DepartmentId { get; set; }
        public string Name { get; set; }
        public DateTime WasAddedDate { get; set; }
        public DateTime? WasChangedDate { get; set; }
        public List<Employee> Employees { get; set; }
    }
}

File: ClientApp/src/app/main/department-management/classes/department.ts
import {Employee} from "./employee";

export class Department {
  departmentId: number = 0;
  name: string = '';
  wasAddedDate: Date | null = null;
  wasChangedDate: Date | null = null;
  employees: Employee[] = [];

  toString(): string {
    return this.name;
  }

  constructor(department: Department|null = null) {
    if(!department) return this;
    this.departmentId = department.departmentId;
    this.name = department.name;
    this.wasChangedDate = department.wasChangedDate;
    this.employees = department.employees;
    this.wasAddedDate = department.wasAddedDate;
  }
}
